src/nectarapi/graphenerpc.py

```
@@ -404,46 +363,71 @@ def get_network(self, props=None):
         else:
             return highest_version_chain
 
-    def _check_for_server_error(self, reply):
+    def _check_for_server_error(self, reply: Dict[str, Any]) -> None:
         """Checks for server error message in reply"""
-        if re.search("Internal Server Error", reply) or re.search("500", reply):
+        reply_str = str(reply)
+        if re.search("Internal Server Error", reply_str) or re.search("500", reply_str):
             raise RPCErrorDoRetry("Internal Server Error")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)